### PR TITLE
State Store Components changes to adhere runtime changes for removing retry options

### LINKED ISSUE #425: Remove retry options from State Apis in components
During review in Dapr maintainers meeting for state building block, it was decided to remove the retry policies for state as retries will be introduced in Dapr in a more cohesive fashion for all building blocks based on community feedback.

Dapr runtime has now the changes to remove the retry policies in State API calls. 

This issue to remove retry options from state apis in components-contrib. This would need changes in the SDKs as well.

This issue tracks overall Dapr retry patterns: #1820

### PR DESCRIPTION
**Pull request by shalabhms**
- Remove retry options from State APIs in components-contrib.
- Replace with a placeholder methods for future request options instead of specific retry options.
- Replace specific retry.go with new request_options to enable any future request options.
- Appropriate changes in the state store components implementations.

## Issue reference

Closes #425 
This issue tracks overall Dapr retry patterns: #1820

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: state/request_options.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		return fmt.Errorf("unrecognized concurrency model '%s'", req.Options.Concurrency)
+	}
+	if req.Options.Consistency != "" && req.Options.Consistency != Strong && req.Options.Consistency != Eventual {
+		return fmt.Errorf("unrecognized consistency model '%s'", req.Options.Consistency)
+	}
+	return nil
+}
+
+// CheckDeleteRequestOptions checks if delete request options use supported keywords
+func CheckDeleteRequestOptions(req *DeleteRequest) error {
```
Comment (yaron2) [inline state/request_options.go:31]: This is a copy paste of `CheckSetRequestOptions`, can we merge?

Comment (shalabhms) [reply state/request_options.go:31]: @yaron2 , these 2 methods are not touched in this changelist as I removed retries only, but I agree these should be changed/merged for sure.
For that, we need to introduce common interface for request. I can introduce new interface and make a change in separate PR. 
Should then I consider that change in different PR ? Please let me know.

Comment (yaron2) [reply state/request_options.go:31]: I would tackle this in this PR.

Comment (shalabhms) [reply state/request_options.go]: done.

**Pull request merged**